Source/Core/VideoCommon/VertexLoaderARM64.cpp

```diff
@@ -456,49 +385,46 @@ void VertexLoaderARM64::GenerateVertexLoader()
   {
-    static const u8 map[8] = {7, 6, 15, 14};
-    const u8 scaling_exponent = map[u32(m_VtxAttr.g0.NormalFormat.Value())];
-    const int limit = m_VtxAttr.g0.NormalElements == NormalComponentCount::NTB ? 3 : 1;
-
-    s32 offset = -1;
-    for (int i = 0; i < limit; i++)
-    {
-      if (!i || m_VtxAttr.g0.NormalIndex3)
-      {
-        int elem_size = GetElementSize(m_VtxAttr.g0.NormalFormat);
-
-        int load_bytes = elem_size * 3;
-        int load_size = GetLoadSize(load_bytes);
+    static constexpr Common::EnumMap<u8, static_cast<ComponentFormat>(7)> SCALE_MAP = {7, 6, 15, 14,
+                                                                                       0, 0, 0,  0};
+    const u8 scaling_exponent = SCALE_MAP[m_VtxAttr.g0.NormalFormat];
 
-        offset = GetAddressImm(CPArray::Normal, m_VtxDesc.low.Normal, EncodeRegTo64(scratch1_reg),
-                               load_size << 3);
+    const int elem_size = GetElementSize(m_VtxAttr.g0.NormalFormat);
+    const int load_bytes = elem_size * 3;
 
-        if (offset == -1)
-          ADD(EncodeRegTo64(scratch1_reg), EncodeRegTo64(scratch1_reg), i * elem_size * 3);
-        else
-          offset += i * elem_size * 3;
-      }
-      int bytes_read = ReadVertex(m_VtxDesc.low.Normal, m_VtxAttr.g0.NormalFormat, 3, 3, true,
-                                  scaling_exponent, &m_native_vtx_decl.normals[i], offset);
+    // Normal
+    auto [reg, offset] = GetVertexAddr(CPArray::Normal, m_VtxDesc.low.Normal);
+    ReadVertex(m_VtxDesc.low.Normal, m_VtxAttr.g0.NormalFormat, 3, 3, true, scaling_exponent,
+               &m_native_vtx_decl.normals[0], reg, offset);
 
-      if (offset == -1)
-        ADD(EncodeRegTo64(scratch1_reg), EncodeRegTo64(scratch1_reg), bytes_read);
-      else
-        offset += bytes_read;
+    if (m_VtxAttr.g0.NormalElements == NormalComponentCount::NTB)
+    {
+      const bool index3 = IsIndexed(m_VtxDesc.low.Normal) && m_VtxAttr.g0.NormalIndex3;
+
+      // Tangent
+      // If in Index3 mode, and indexed components are used, replace the index with a new index.
+      if (index3)
+        std::tie(reg, offset) = GetVertexAddr(CPArray::Normal, m_VtxDesc.low.Normal);
+      // The tangent comes after the normal; even in index3 mode, an extra offset of load_bytes is
+      // applied. Note that this is different from adding 1 to the index, as the stride for indices
+      // may be different from the size of the tangent itself.
+      ReadVertex(m_VtxDesc.low.Normal, m_VtxAttr.g0.NormalFormat, 3, 3, true, scaling_exponent,
+                 &m_native_vtx_decl.normals[1], reg, offset + load_bytes);
+      // Binormal
+      if (index3)
+        std::tie(reg, offset) = GetVertexAddr(CPArray::Normal, m_VtxDesc.low.Normal);
+      ReadVertex(m_VtxDesc.low.Normal, m_VtxAttr.g0.NormalFormat, 3, 3, true, scaling_exponent,
+                 &m_native_vtx_decl.normals[2], reg, offset + load_bytes * 2);
```

Comment:
In the x64 code you're adding elem_size * 2, but here you're adding load_bytes * 2. Is this intentional?

Reply:
Just bad variable naming; they're the same value:

https://github.com/dolphin-emu/dolphin/blob/1e52eb1959fb1d53c0b05467a81f9a245bf56940/Source/Core/VideoCommon/VertexLoaderX64.cpp#L473
https://github.com/dolphin-emu/dolphin/blob/1e52eb1959fb1d53c0b05467a81f9a245bf56940/Source/Core/VideoCommon/VertexLoaderARM64.cpp#L402-L403

I'll change the x64 one.